src/CMakeLists.txt

```diff
@@ -180,6 +180,21 @@ endif(ENABLE_REDIS)
 find_package(SQLite3 REQUIRED)
 find_package(Json REQUIRED)
 
+OPTION(ENABLE_SPATIAL "Enable SpatialIndex AreaStore-backend" TRUE)
+set(USE_SPATIAL FALSE)
+
+if(ENABLE_SPATIAL)
+	find_library(SPATIAL_LIBRARY spatialindex)
+	find_path(SPATIAL_INCLUDE_DIR SpatialIndex.h PATH_SUFFIXES spatialindex)
```

Comment:
I think you should use `find_path(SPATIAL_INCLUDE_DIR spatialindex/SpatialIndex.h)` and change the includes appropriately to keep things organized and avoid conflicts.